indy_node/server/domain_req_handler.py

```diff
@@ -22,9 +23,172 @@
 logger = getlogger()
 
 
+class RevocationStrategy:
+
+    def __init__(self, state):
+        self.state = state
+        self.author_did = None
+        self.revoc_reg_def_id = None
+        self.req_id = None
+
+    def set_parameters_from_txn(self, author_did, revoc_reg_def_id, req_id):
+        self.author_did = author_did
+        self.revoc_reg_def_id = revoc_reg_def_id
+        self.req_id = req_id
+
+    def set_strategy(self, strategy):
+        self.strategy = strategy
+
+    def set_to_state(self, txn):
+        assert self.strategy
+        self.strategy.write(txn)
+
+    def validate_txn(self, current_entry, req: Request):
+        self.set_parameters_from_txn(author_did=req.identifier,
+                                     revoc_reg_def_id=req.operation[REVOC_REG_DEF_ID],
+                                     req_id=req.reqId)
+        # General checks for all Revocation entries
+        operation = req.operation
+        value_from_txn = operation.get(VALUE)
+        issued_from_txn = value_from_txn.get(ISSUED)
+        revoked_from_txn = value_from_txn.get(REVOKED)
+        intersection = set(issued_from_txn).intersection(set(revoked_from_txn))
+        if len(intersection) > 0:
+            raise InvalidClientRequest(self.author_did,
+                                       self.req_id,
+                                       "Indices {} are existed in issued and revoked list".format(intersection))
+        if current_entry is None:
+            return None
+        value_from_state = current_entry.get(VALUE)
+        assert value_from_state
+        current_accum = value_from_state.get(ACCUM)
+        if current_accum != value_from_txn.get(PREV_ACCUM):
+            raise InvalidClientRequest(self.author_did,
+                                       self.req_id,
+                                       "Accum value from state: {} "
+                                       "do not equal with {} value "
```

Comment:
It's better to say:
The current accumulator value {} must be equal to the last accumulator value {} in transaction.